pkg/config/validator.go

```diff
@@ -72,19 +73,35 @@ type (
 	// AdapterToAspectMapper returns the set of aspect kinds implemented by
 	// the given builder.
 	AdapterToAspectMapper func(builder string) KindSet
+
+	// BuilderInfoFinder is used to find specific handlers BuilderInfo for configuration.
+	BuilderInfoFinder func(name string) (*adapter.BuilderInfo, bool)
+
+	// SetupHandlerFn is used to configure handler implementation with Types associated with all the templates that
+	// it supports.
+	SetupHandlerFn func(actions []*pb.Action, constructors map[string]*pb.Constructor,
+		handlers map[string]*HandlerBuilderInfo, tmplRepo template.Repository, expr expr.TypeChecker, df expr.AttributeDescriptorFinder) error
 )
 
 // newValidator returns a validator given component validators.
 func newValidator(managerFinder AspectValidatorFinder, adapterFinder BuilderValidatorFinder,
+	builderInfoFinder BuilderInfoFinder, setupHandlerFn SetupHandlerFn, templateRepo template.Repository,
 	findAspects AdapterToAspectMapper, strict bool, typeChecker expr.TypeChecker) *validator {
 	return &validator{
-		managerFinder: managerFinder,
-		adapterFinder: adapterFinder,
-		findAspects:   findAspects,
-		strict:        strict,
-		typeChecker:   typeChecker,
+		managerFinder:   managerFinder,
```

Comment:
To be clear, doesn't need to happen in this PR. But we need to revisit this file.